src/MissionManager/QGCMapPolygon.cc

```diff
@@ -622,3 +623,14 @@ void QGCMapPolygon::setShowAltColor(bool showAltColor){
         emit showAltColorChanged(showAltColor);
     }
 }
+
+void QGCMapPolygon::selectVertex(int index)
+{
+    if(0 <= index && index < count() && index != _selectedVertexIndex) {
```

Comment:
Can you break this up so you can set selection to -1 when you get a index out of bounds and spit out a qWarning about it as well? That indicates in internal code error which should be flagged.

Reply:
Good idea, done: https://github.com/mavlink/qgroundcontrol/pull/8813/commits/ce27d8fc0736710ddf9a446b0cba9b5761785d6f